### PR TITLE
additional cleanup from PR 5462

### PR DESCRIPTION
**Pull request by Goober5000**
Some streamlining and constifying of the readyroom popup list and associated functions.

Followup to #5462.

### CODE REVIEW

File: code/hud/hudconfig.cpp [hunk 1]
```diff
@@ -1708,10 +1708,9 @@ void hud_config_color_load(const char *name)
 {
 	int idx;
 	char str[1024];
-	char *fname;
 	ubyte r, g, b, a;
 
-	fname = cf_add_ext(name, ".hcf");
+	auto fname = cf_add_ext(name, ".hcf");
```
Comment (jg18) [inline code/hud/hudconfig.cpp:1713]: `const auto`? Not required but makes it easier to tell it's `const`.

Comment (jg18) [reply code/hud/hudconfig.cpp:1713]: IMHO `const char *` is easier to read and not much more text than `auto` but seems like the latter is SCP standard.

Comment (Goober5000) [reply code/hud/hudconfig.cpp:1713]: At one point our CI would complain about long type names where `auto` would suffice, but that's currently disabled.  I'm fine with putting `const char *` here.

**Review by wookieejedi — APPROVED** [review]
Works in my tests and PR is straightforward.

**Pull request merged**